### PR TITLE
fix Issue 24274 - [REG master] ImportC: unrecognized C initializer wi…

### PR DESCRIPTION
**Pull request by WalterBright**
…th array in struct

### CODE REVIEW

File: compiler/src/dmd/initsem.d [hunk 1]
```diff
@@ -833,34 +830,62 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, ref Typ
...
+                                    ix = inix;
+                            }
+                        }
+                    }
+
+                    if (auto cix = ix.isCInitializer())
+                    {
+                        /* ImportC loses the structure from anonymous structs, but this is retained
+                         * by the initializer syntax. if a CInitializer has a Designator, it is probably
+                         * a nested anonymous struct
```
Comment (dkorpel) [inline compiler/src/dmd/initsem.d:865]: Is there a way for dmd to retain info about anonymous structs? The compiler shouldn't rely on guess work.

**Review by dkorpel — APPROVED** [review]

**Comment by dkorpel** [comment]
Next bug: https://issues.dlang.org/show_bug.cgi?id=24277

**Comment by dkorpel** [comment]
That *may* be the last one I'll file :slightly_smiling_face: , adding a workaround for it makes ImportC finally able to compile tree-sitter and pass all my unit tests.

File: compiler/src/dmd/initsem.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/initsem.d:865]: I have some ideas, but wanted to at least get this working.

**Comment by WalterBright** [comment]
The problem is based on anonymous structs. You can generally avoid it by not using { } for the anonymous initializers, or by naming the structs.

**Pull request merged**